Installation issue: parallel-hashmap

### Steps to reproduce the issue

```console
$ spack install parallel-hashmap
```


### Error message

<details><summary>Error message</summary>

<pre>
5 errors found in build log:
     16    fatal: unable to access 'https://github.com/google/googletest.git/': Empty reply from server
     17    Cloning into 'googletest-src'...
     18    fatal: unable to access 'https://github.com/google/googletest.git/': Empty reply from server
     19    Cloning into 'googletest-src'...
     20    fatal: unable to access 'https://github.com/google/googletest.git/': Failed to connect to github.com port 443:
            Connection timed out
     21    Had to git clone more than once: 3 times.
  >> 22    CMake Error at googletest-download/googletest-prefix/tmp/googletest-gitclone.cmake:50 (message):
     23      Failed to clone repository: 'https://github.com/google/googletest.git'
     24
     25
  >> 26    gmake[2]: *** [CMakeFiles/googletest.dir/build.make:99: googletest-prefix/src/googletest-stamp/googletest-down
           load] Error 1
  >> 27    gmake[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/googletest.dir/all] Error 2
  >> 28    gmake: *** [Makefile:91: all] Error 2
  >> 29    CMake Error at cmake/DownloadGTest.cmake:27 (message):
     30      Build step for googletest failed: 2
     31    Call Stack (most recent call first):
     32      CMakeLists.txt:98 (include)
     33
     34
     35    -- Configuring incomplete, errors occurred!
</pre></details>
I tried to install `parallel-hashmap` on the cluster by `spack mirror`, where accessing the Internet is quite limited. But during `Executing phase: 'cmake'`, the software will try to download `googletest` from github, instead of using `googletest` from spack `mirror`.

### Information on your system

* **Spack:** 0.23.0.dev0 (c29652580a8d719b25afa80af98ccc9799ca4f5b)
* **Python:** 3.7.9
* **Platform:** linux-kylinV10-aarch64


### Additional information

[spack-build-out.txt](https://github.com/user-attachments/files/16735947/spack-build-out.txt)


### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/parallel-hashmap/package.py
+++ b/var/spack/repos/builtin/packages/parallel-hashmap/package.py
@@ -18,7 +18,7 @@ class ParallelHashmap(CMakePackage):
     version("1.3.12", sha256="0cc203144321924cfbfcc401f42d8204c0dd24e2760c7a1c091baa16d9777c08")
     version("1.3.11", sha256="0515a681bfb24207013786a7737e9d8561302e656689d8a65ea480bbabab460f")
 
-    depends_on("cxx", type="build")  # generated
+    depends_on("cxx", type="build")
 
     depends_on("cmake@3.8:", type="build")
 

--- a/var/spack/repos/builtin/packages/parallel-hashmap/package.py
+++ b/var/spack/repos/builtin/packages/parallel-hashmap/package.py
@@ -18,8 +18,8 @@ class ParallelHashmap(CMakePackage):
     version("1.3.12", sha256="0cc203144321924cfbfcc401f42d8204c0dd24e2760c7a1c091baa16d9777c08")
     version("1.3.11", sha256="0515a681bfb24207013786a7737e9d8561302e656689d8a65ea480bbabab460f")
 
-    depends_on("cxx", type="build")
 
+    depends_on("cxx", type="build")
     depends_on("cmake@3.8:", type="build")
 
     patch("pthread.patch")

--- a/var/spack/repos/builtin/packages/parallel-hashmap/package.py
+++ b/var/spack/repos/builtin/packages/parallel-hashmap/package.py
@@ -15,6 +15,7 @@ class ParallelHashmap(CMakePackage):
 
     license("Apache-2.0")
 
+    version("1.3.12", sha256="0cc203144321924cfbfcc401f42d8204c0dd24e2760c7a1c091baa16d9777c08")
     version("1.3.11", sha256="0515a681bfb24207013786a7737e9d8561302e656689d8a65ea480bbabab460f")
 
     depends_on("cxx", type="build")  # generated

--- a/var/spack/repos/builtin/packages/parallel-hashmap/package.py
+++ b/var/spack/repos/builtin/packages/parallel-hashmap/package.py
@@ -31,7 +31,7 @@ class ParallelHashmap(CMakePackage):
 
         args = [
             from_variant("PHMAP_BUILD_EXAMPLES", "examples"),
-            define("PHMAP_BUILD_TESTS", False) # disable due to vendored gtest
+            define("PHMAP_BUILD_TESTS", False),  # disable due to vendored gtest
         ]
 
         return args

--- a/var/spack/repos/builtin/packages/parallel-hashmap/package.py
+++ b/var/spack/repos/builtin/packages/parallel-hashmap/package.py
@@ -18,8 +18,17 @@ class ParallelHashmap(CMakePackage):
     version("1.3.12", sha256="0cc203144321924cfbfcc401f42d8204c0dd24e2760c7a1c091baa16d9777c08")
     version("1.3.11", sha256="0515a681bfb24207013786a7737e9d8561302e656689d8a65ea480bbabab460f")
 
+    variant("examples", description="Build examples", default=False)
 
     depends_on("cxx", type="build")
     depends_on("cmake@3.8:", type="build")
 
     patch("pthread.patch")
+
+    def cmake_args(self):
+        from_variant = self.define_from_variant
+        args = [
+            from_variant("PHMAP_BUILD_EXAMPLES", "examples"),
+        ]
+
+        return args

--- a/var/spack/repos/builtin/packages/parallel-hashmap/package.py
+++ b/var/spack/repos/builtin/packages/parallel-hashmap/package.py
@@ -26,12 +26,9 @@ class ParallelHashmap(CMakePackage):
     patch("pthread.patch")
 
     def cmake_args(self):
-        define = self.define
-        from_variant = self.define_from_variant
-
         args = [
-            from_variant("PHMAP_BUILD_EXAMPLES", "examples"),
-            define("PHMAP_BUILD_TESTS", False),  # disable due to vendored gtest
+            self.define_from_variant("PHMAP_BUILD_EXAMPLES", "examples"),
+            self.define("PHMAP_BUILD_TESTS", False),  # disable due to vendored gtest
         ]
 
         return args

--- a/var/spack/repos/builtin/packages/parallel-hashmap/package.py
+++ b/var/spack/repos/builtin/packages/parallel-hashmap/package.py
@@ -26,9 +26,12 @@ class ParallelHashmap(CMakePackage):
     patch("pthread.patch")
 
     def cmake_args(self):
+        define = self.define
         from_variant = self.define_from_variant
+
         args = [
             from_variant("PHMAP_BUILD_EXAMPLES", "examples"),
+            define("PHMAP_BUILD_TESTS", False) # disable due to vendored gtest
         ]
 
         return args

